turbine/src/retransmit_stage.rs

```diff
@@ -846,73 +847,45 @@ mod tests {
 
     #[test]
     fn test_already_received() {
-        let slot = 1;
-        let index = 5;
-        let version = 0x40;
         let keypair = get_keypair();
-        let mut shred = Shred::new_from_data(
-            slot,
-            index,
+        let entries = create_ticks(10, 1, Hash::new_unique());
+        let shredder = Shredder::new(0, 0, 1, 0).unwrap();
+        let rsc = ReedSolomonCache::default();
+        let (shreds_data, shreds_code) = shredder.entries_to_shreds(
+            &keypair,
+            &entries,
+            true,
+            // chained_merkle_root
+            Some(Hash::new_from_array(rand::thread_rng().gen())),
             0,
-            &[],
-            ShredFlags::LAST_SHRED_IN_SLOT,
-            0,
-            version,
             0,
+            true,
+            &rsc,
+            &mut ProcessShredsStats::default(),
         );
-        shred.sign(&keypair);
+        let shred = shreds_data[0].clone();
         let mut rng = ChaChaRng::from_seed([0xa5; 32]);
         let shred_deduper = ShredDeduper::<2>::new(&mut rng, /*num_bits:*/ 640_007);
-        // unique shred for (1, 5) should pass
+        // unique shred should pass
         assert!(!shred_deduper.dedup(shred.id(), shred.payload(), MAX_DUPLICATE_COUNT));
-        // duplicate shred for (1, 5) blocked
+        // duplicate shred blocked
         assert!(shred_deduper.dedup(shred.id(), shred.payload(), MAX_DUPLICATE_COUNT));
-        let mut shred = Shred::new_from_data(
-            slot,
-            index,
-            2,
-            &[],
-            ShredFlags::LAST_SHRED_IN_SLOT,
-            0,
-            version,
-            0,
-        );
-        shred.sign(&keypair);
-        // first duplicate shred for (1, 5) passed
+        let shred = shreds_data.last().unwrap().clone();
```

Comment:
this one is not a duplicate though because the index has changed.

I think we should bring back the test cases where we use the same slot/index but just stomp the parent slot

Reply:
we do not have the relevant setters in merkle shred's public API. I'll make a workaround.